src/www/ui/api/Controllers/UploadTreeController.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/*
+ SPDX-FileCopyrightText: © 2023 Samuel Dushimimana <dushsam100@gmail.com>
+
+ SPDX-License-Identifier: GPL-2.0-only
+*/
+
+/**
+ * @file
+ * @brief Controller for uploadtree queries
+ */
+
+namespace Fossology\UI\Api\Controllers;
+
+use Fossology\UI\Api\Helper\ResponseHelper;
+use Fossology\UI\Api\Models\Info;
+use Fossology\UI\Api\Models\InfoType;
+use Psr\Http\Message\ServerRequestInterface;
+
+
+/**
+ * @class UploadTreeController
+ * @brief Controller for UploadTree model
+ */
+class UploadTreeController extends RestController
+{
+
+  /**
+   * Get the contents of a specific file
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function viewLicenseFile($request, $response, $args)
+  {
+    $upload_pk = intval($args['id']);
+    $item_pk = intval($args['itemId']);
+    $_GET['upload'] = $upload_pk;
+    $_GET['item'] = $item_pk;
+
+    $uploadDao = $this->restHelper->getUploadDao();
+    $returnVal = null;
+
+    if (!$this->dbHelper->doesIdExist("upload", "upload_pk", $upload_pk)) {
+      $returnVal = new Info(404, "Upload does not exist", InfoType::ERROR);
+    } else if (!$this->dbHelper->doesIdExist($uploadDao->getUploadtreeTableName($upload_pk), "uploadtree_pk", $item_pk)) {
+      $returnVal = new Info(404, "Item does not exist", InfoType::ERROR);
+    }
+
+    if ($returnVal !== null) {
+      return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+    }
+
+    $view = $this->restHelper->getPlugin('view');
+
+    list($pageMenu, $returnVal) = $view->getView(NULL, "license", 0, "", array(), false, true);
+    return $response->withJson(strip_tags($returnVal), 200);
```

Comment:
Would suggest to send plain text rather than a JSON.
```suggestion
    $response->getBody()->write(strip_tags($returnVal));
    return $response->withHeader("Content-Type", "text/plain")
      ->withHeader("Cache-Control", "max-age=1296000, must-revalidate")
      ->withHeader("Etag", md5($response->getBody()));
```

Reply:
Thanks, Done 

![image](https://github.com/fossology/fossology/assets/66276301/93ce26f1-6186-486e-8a2d-ba9b26dfd3a0)